### DISCUSSION TITLE
Adding cost linting

### DISCUSSION THREAD

**Discussion by swarner1033** [terraform-linters/tflint-ruleset-aws#553] [Ideas]
Would tflint be open to PRs that added more cost governance functionality? These would be rules like: S3 bucket must have a lifecycle or EC2 instance must be of the latest generation.

**Comment by bendrucker** [comment]
Rules can be proposed for any reason. The cases you've offered aren't particularly strong indications that the general category of "cost governance" will be fruitful:

* S3 buckets don't inherently merit lifecycle rules.
* Using the latest generation of instances is a potentially good suggestion, for many reasons other than cost. 

I think you'll end up finding that cost recommendations typically require context that static configuration analysis can't see.
